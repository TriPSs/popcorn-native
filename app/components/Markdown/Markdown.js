import React from 'react'
import PropTypes from 'prop-types'
import MD from 'react-native-markdown-renderer'

import dimensions from 'modules/dimensions'

import Typography from '../Typography'

export const style = {

  heading1: Typography.getTextStyle({
    variant: 'headline5',
    asObject: true,
  }),

  heading2: Typography.getTextStyle({
    variant: 'headline6',
    asObject: true,
  }),

  heading3: Typography.getTextStyle({
    variant: 'subtitle2',
    asObject: true,
  }),

  text: Typography.getTextStyle({
    variant: 'body2',
    asObject: true,
  }),

  listUnorderedItemIcon: {
    lineHeight: dimensions.UNIT * 4,
    marginLeft: dimensions.UNIT,
    marginRight: dimensions.UNIT,
  },

  listItem: {
    flex: 1,
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'center',
    maxHeight: dimensions.UNIT * 4,
  },

}

export const Markdown = ({ children }) => (
  <MD style={style}>
    {children}
  </MD>
)

Markdown.propTypes = {
  children: PropTypes.string.isRequired,
}

export default Markdown
