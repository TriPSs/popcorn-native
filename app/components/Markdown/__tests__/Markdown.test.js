import React from 'react'
import { shallow } from 'enzyme'

import Markdown from '../Markdown'

describe('(Component) Markdown', () => {

  it('should render', () => {
    const result = shallow(
      <Markdown>
        ## Test markdown

        - test
      </Markdown>,
    )

    expect(result).toMatchSnapshot()
  })

})
