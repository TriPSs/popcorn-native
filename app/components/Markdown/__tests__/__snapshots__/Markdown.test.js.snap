// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Markdown should render with defaults 1`] = `
<_default
  style={
    Object {
      "heading1": Object {
        "color": "#ffffff",
        "fontFamily": "Quicksand-Regular",
        "fontSize": 24,
        "letterSpacing": 0.18,
        "lineHeight": 24,
        "opacity": 0.87,
      },
      "heading2": Object {
        "color": "#ffffff",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 20,
        "letterSpacing": 0.15,
        "lineHeight": 24,
        "opacity": 0.87,
      },
      "heading3": Object {
        "color": "#ffffff",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 14,
        "letterSpacing": 0.1,
        "lineHeight": 24,
        "opacity": 0.87,
      },
      "listItem": Object {
        "alignItems": "center",
        "flex": 1,
        "flexWrap": "wrap",
        "justifyContent": "center",
        "maxHeight": 32,
      },
      "listUnorderedItemIcon": Object {
        "lineHeight": 32,
        "marginLeft": 8,
        "marginRight": 8,
      },
      "text": Object {
        "color": "#ffffff",
        "fontFamily": "Quicksand-Regular",
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 0.87,
      },
    }
  }
>
  ## Markdown test
</_default>
`;
